DEV-2107 Calculate title

spacid opened the pull request:
Although `/VIAA/dc_title` is mandatory, it might be absent from some
sidecars of some CPs. Implement a kind of fallback mechanism based on
the previous borndigital logic.

Use in order:
 - `/VIAA/dc_title`
  -`/VIAA/dc_titles/*[1]`
 - `/VIAA/dc_description_short`

spacid committed:
DEV-2107 Calculate title

Although `/VIAA/dc_title` is mandatory, it might be absent from some
sidecars of some CPs. Implement a kind of fallback mechanism based on
the previous borndigital logic.

Use in order:
 - `/VIAA/dc_title`
  -`/VIAA/dc_titles/*[1]`
 - `/VIAA/dc_description_short`

spacid commented:
The order in the transformed metadata is changed: `dcterms:title will be on top.`

RudolfDG commented on app/resources/dcterms.xslt:
```diff
@@ -887,11 +894,29 @@
+                <xsl:value-of select="$titles_first" />
+            </xsl:element>
+        </xsl:if>
+        <xsl:if test="$description_short and not($titles_first and $title)">
```
`$description_short and not($titles_first and $title)` moet `$description_short and not($titles_first or $title)` zijn. Anders gaat er 2 keer dcterms:title aanwezig zijn als bvb title en description_short ingevuld zijn.

spacid committed:
DEV-2107 Fix in title calculation

The field `dc_description_short` should only be used if there is no value
for both `dc_title` as `dc_titles`.
